### PR TITLE
[Network] az network private-endpoint create: add more declaration of '--subnet' and '--private-connection-resource-id'

### PR DESCRIPTION
**Pull request by msyyc**
Close https://github.com/Azure/azure-cli/issues/15944

--

### CODE REVIEW

**Comment by yonzhan** [comment]
network

**Review by haroldrandom — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 1]
```diff
@@ -812,10 +812,10 @@ def load_arguments(self, _):
     with self.argument_context('network private-endpoint') as c:
         c.argument('private_endpoint_name', private_endpoint_name, options_list=['--name', '-n'])
         c.argument('location', get_location_type(self.cli_ctx), validator=get_default_location_from_resource_group)
-        subnet_help = get_folded_parameter_help_string('subnet', other_required_option='--vnet-name')
+        subnet_help = get_folded_parameter_help_string('subnet', other_required_option='--vnet-name') + ' and do not specify the --vnet-name'
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:815]: Can we also validate this logic in validator as well?

Comment (msyyc) [reply src/azure-cli/azure/cli/command_modules/network/_params.py:815]: we indeed tell the exclusion in 'help message', but not in the same message block. Maybe the issue owner think it should be more obvious.

**Pull request merged**